app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -43,45 +60,62 @@ class DroidCvPattern(
 
     override fun close() {
         if (ownsMat) {
-            mat?.release()
+            mat.release()
         }
-
-        mat = null
+        alpha?.release()
     }
 
-    private fun resize(target: Mat, size: Size) {
+    private fun resize(source: Mat, target: Mat, size: Size) {
         Imgproc.resize(
-            mat, target,
+            source, target,
             CvSize(size.width.toDouble(), size.height.toDouble()),
             0.0, 0.0, Imgproc.INTER_AREA
         )
     }
 
     override fun resize(size: Size): IPattern {
-        val result = Mat()
-        resize(result, size)
-        return DroidCvPattern(result).tag(tag)
+        val resizedMat = Mat().apply { resize(mat, this, size) }
+        val resizedAlpha = alpha?.let {
+            Mat().apply { resize(it, this, size) }
+        }
+
+        return DroidCvPattern(resizedMat, resizedAlpha).tag(tag)
     }
 
     override fun resize(target: IPattern, size: Size) {
         if (target is DroidCvPattern) {
-            resize(target.mat!!, size)
+            resize(mat, target.mat, size)
+            alpha?.let {
+                if (target.alpha == null) {
+                    target.alpha = Mat()
+                }
+                resize(it, target.alpha!!, size)
```

Comment:
```suggestion
                target.alpha?.let { targetAlpha ->
                    resize(it, targetAlpha, size)
                }
```

I hate `null` exceptions => I hate `!!`.

Reply:
A null exception could never happen because I'm changing it to a non-null value in that case.

I don't really like using nested `let`s. How about this instead:
```kt
alpha?.let { originalAlpha ->
    target.alpha = (target.alpha ?: Mat()).apply {
        resize(originalAlpha, this, size)
    }
}
 ```

Reply:
I'm fine with your suggestion.

> could never happen

you never know how much the code might change in the future.